src/Nethermind/Nethermind.Network/PeerManager.cs

```diff
@@ -96,8 +96,9 @@ public class PeerManager : IPeerManager
 
         public IReadOnlyCollection<Peer> ActivePeers => _activePeers.Values.ToList().AsReadOnly();
         public IReadOnlyCollection<Peer> CandidatePeers => _peerPool.CandidatePeers.ToList();
+        public IReadOnlyCollection<Peer> ConnectedPeers => _activePeers.Values.Where(IsConnected).ToList().AsReadOnly();
```

Comment:
Maybe just IEnumerable? Not sure myself.

Reply:
better follow the rules as in others, IEnumerable would be dangerous here because we have multiple threads operating on some of those lists